src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -231,16 +361,27 @@ bool AutomatedRssDownloader::isRssDownloaderEnabled() const
     return ui->checkEnableDownloader->isChecked();
 }
 
-void AutomatedRssDownloader::updateRuleDefinitionBox()
+void AutomatedRssDownloader::updateRuleDefinitionBox(QListWidgetItem *selected)
 {
+    disconnectRuleFeedSlots();
+
     qDebug() << Q_FUNC_INFO;
     // Save previous rule first
     saveEditedRule();
     // Update rule definition box
     const QList<QListWidgetItem *> selection = ui->listRules->selectedItems();
-    if (selection.count() == 1) {
-        m_editedRule = selection.first();
-        Rss::DownloadRulePtr rule = getCurrentRule();
+
+    if (!selected && (selection.count() == 1))
```

Comment:
... I find a little hard to follow what the code is trying to do, is it possible to refactor it?
I mean there are 2 boolean conditions so there are 4 outcomes, reading the code I cannot conclude that all possibilities are covered.
Or maybe: `!selection.empty()`?